src/controllers/returnTrip.js

```diff
@@ -0,0 +1,83 @@
+import { User, Trip, Accomodation } from '../database/models';
+
+class returnTripController {
+  async getAll(req, res) {
+    const trips = await Trip.findAll({
+      include: [{ model: Accomodation }, { model: User }]
+    });
+    res.send(trips);
+  }
+  async createTrip(req, res) {
+    const {
+      departFrom,
+      departTo,
+      departDate,
+      returnDate,
+      travelReasons,
+      managerEmail,
+      accomodationId
+    } = req.body;
+    const getAccomodation = await Accomodation.findOne({
+      where: { id: accomodationId }
+    });
+    const newTrip = {
+      departFrom,
+      departTo,
+      departDate,
+      returnDate,
+      travelReasons,
+      managerEmail,
+      approvalStatus: false,
+      accomodationId: getAccomodation.id,
+      userId: req.user.id
+    };
+    await Trip.create(newTrip);
+    res.status(200).json({
+      status: 'ok',
+      msg: 'Trip requested success',
+      trip: newTrip
+    });
+  }
+  async deleteTrip(req, res) {
+    const trip = await Trip.findOne({ where: { id: req.params.id } });
+    await trip.destroy();
```

Comment:
Can you first check if the trip is available before destroying it?
It can prevent these kinds of errors 👇
<img width="458" alt="Screen Shot 2020-10-06 at 14 09 51" src="https://user-images.githubusercontent.com/34744026/95199785-adcd1e80-07dd-11eb-9dbb-ed06c43b5388.png">

Reply:
This applies to accommodation too.

Reply:
i have correct it